### PR TITLE
Project 12 Movies:  Ann-Sofi Jönsson & Darcy Winterhill

### PR DESCRIPTION
**Pull request by Ann-Sofi**
Link to deployed project:

https://movie-site-2021.netlify.app/

### CODE REVIEW

File: code/public/index.html [hunk 1]
```diff
@@ -13,7 +14,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
```
Comment (rrpersson) [inline code/public/index.html:16]: Perhaps you could remove the comments from the production build!

File: code/src/reusables/urls.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const API_KEY = 'e90c1322becf61268c0c0fea570c7099';
+export const MOVIE_URL = `https://api.themoviedb.org/3/movie/popular?api_key=${API_KEY}&language=en-US&page=1`;
+
+
+
```
Comment (rrpersson) [inline code/src/reusables/urls.js:5]: Nice use of an external file with urls

File: code/src/pages/MovieDetails.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+export const MovieDetails = () => {
+
+  const {id} = useParams();
+  const [movieDetails, setMovieDetails] = useState()
+
+  useEffect(() => {
+    fetch(`https://api.themoviedb.org/3/movie/${id}?api_key=e90c1322becf61268c0c0fea570c7099&language=en-US`)
+    .then((res) => res.json())
+    .then((json) => setMovieDetails(json))
+  }, [id])
```
Comment (rrpersson) [inline code/src/pages/MovieDetails.js:13]: Perhaps this link could be moved to your URLs file! :)

File: code/src/components/MovieList.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React, { useEffect, useState} from 'react'
+
+import { MovieCard } from './MovieCard'
+import { MOVIE_URL } from '../reusables/urls'
```
Comment (rrpersson) [inline code/src/components/MovieList.js:4]: Good job divinding the imports!

File: code/src/App.js [hunk 5]
```diff
@@ -1,9 +1,23 @@
...
+          <MovieList />
+        </Route>
+        <Route path='/movie/:id'>
+          <MovieDetails />
+        </Route>
+      </Switch>
+    </BrowserRouter>
     </div>
   )
 }
```
Comment (rrpersson) [inline code/src/App.js:23]: You app file looks clean and it's easy to follow along and see what's going on

File: code/src/components/MovieCard.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+          <Link to={`/movie/${id}`}>
+            <img src={`https://image.tmdb.org/t/p/w342${poster_path}`} alt={title} />
+            <div className='details'>
+                <h1>{title}</h1>
+                <p>Released {release_date}</p>
+            </div>
+          </Link>
+      </>
+    )
+}
```
Comment (rrpersson) [inline code/src/components/MovieCard.js:17]: It's easy to understand the component and what it does. Well done!

File: code/src/pages/MovieDetails.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  return (
+    <article className='detail-page'>
+
+        <Link to='/' className='back-button'>
+            <i className="fas fa-chevron-circle-left"></i>
+            <p>Movies</p>
+        </Link>
+
+        <div className='background' style={{backgroundImage: ` linear-gradient(rgba(0, 0, 0, 0) 70%, rgb(0, 0, 0) 100%), url(https://image.tmdb.org/t/p/w1280${movieDetails.backdrop_path})` }} alt={movieDetails.title}>
```
Comment (rrpersson) [inline code/src/pages/MovieDetails.js:27]: Very nice effect. It's very close to the original! Nice job!

File: code/src/pages/MovieDetails.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  const [movieDetails, setMovieDetails] = useState()
+
+  useEffect(() => {
+    fetch(`https://api.themoviedb.org/3/movie/${id}?api_key=e90c1322becf61268c0c0fea570c7099&language=en-US`)
+    .then((res) => res.json())
+    .then((json) => setMovieDetails(json))
+  }, [id])
+
+if (movieDetails === undefined) {
+  return <></>
```
Comment (rrpersson) [inline code/src/pages/MovieDetails.js:16]: Good use of a simple if statement

**Review by rrpersson — APPROVED** [review]
Your project looks nice. It looks very close to the original which is great. Very well done! ⭐🌟

**Review by KarolCislowski — APPROVED** [review]
Great Project 🌈

**Pull request closed without merging**